### PR TITLE
Fix spelling and grammar mistakes and improve sentence clarity

### PR DESCRIPTION
**Pull request by terrchen**
I saw this project on Hacker News, and I thought I would run it across my Markdown analyzer and found some errors you might want to consider merging.  To learn more about why the changes were made, you can review the AI analysis for this README at https://app.gitsense.com/microsoft/data-formulator?doc=1868de2e176c652&highlight=conflicts&other-models=Claude+3.5+Sonnet%2CGPT-4o%2CLlama+3.1+70B

Note, the highlighted text means there were conflicts between models.

### CODE REVIEW

**Comment by danmarshall** [comment]
Thanks! Would you mind undoing the changes to the boilerplate sections?

**Comment by Chenglong-MS** [comment]
I think we can merge this into dev, and then merge to main a little later -- also wants to update the deployment criteria

**Comment by terrchen** [comment]
> Thanks! Would you mind undoing the changes to the boilerplate sections?

Do you mean the license information?

File: README.md [hunk 1]
```diff
@@ -140,7 +140,7 @@ to provide a CLA and decorate the PR appropriately (e.g., label, comment). Simpl
 instructions provided by the bot. You will only need to do this once across all repositories using our CLA.
 
 This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
-For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/)
+For more information, see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/)
```
Comment (danmarshall) [inline README.md]: Here

File: README.md [hunk 2]
```diff
@@ -149,4 +149,4 @@ This project may contain trademarks or logos for projects, products, or services
 trademarks or logos is subject to and must follow 
 [Microsoft's Trademark & Brand Guidelines](https://www.microsoft.com/en-us/legal/intellectualproperty/trademarks/usage/general).
 Use of Microsoft trademarks or logos in modified versions of this project must not cause confusion or imply Microsoft sponsorship.
-Any use of third-party trademarks or logos are subject to those third-party's policies.
+Any use of third-party trademarks or logos is subject to those third parties' policies.
```
Comment (danmarshall) [inline README.md]: And here

Comment (terrchen) [reply README.md]: Sure.  I am on the move right now so I won't be able to undo the change this until later today.

Comment (terrchen) [reply README.md]: updated

**Comment by danmarshall** [comment]
Thanks!

**Pull request merged**